### DISCUSSION TITLE
Invocation of "subscription" cannot occur because arguments do not match any registered target methods.

### DISCUSSION THREAD

**Discussion by jchapmantcg** [microsoft/vs-streamjsonrpc#792] [Q&A]
Hello,

I'm trying to get familiar with this library by writing an Api wrapper. I've been able to get some basic request/responses working with the `Invoke` methods and now I'm trying to connect a delegate to a notification.

The Api I'm working with allows you to "subscribe" to data changes. 

First questions is, how do I properly setup the delegate to handle these notifications?

```
// This works to setup the subscription properly
await rpc.InvokeWithParameterObjectAsync("public/subscribe", new { channels = new[] { "BTC" } });

// This seems to be the proper method to use since I get this error message when a notification is received
"Invocation of "subscription" cannot occur because arguments do not match any registered target methods."
rpc.AddLocalRpcMethod("subscription", () =>
{
	Console.WriteLine($"Notification received!");
});
```

Next question is, since this Api supports tons of data to subscribe to, how do I connect a different delegate to each?
```
// This should go to one delegate
await rpc.InvokeWithParameterObjectAsync("public/subscribe", new { channels = new[] { "BTC" } });

// While this should use another
await rpc.InvokeWithParameterObjectAsync("public/subscribe", new { channels = new[] { "ETH" } });
```

**Comment by AArnott** [comment]
It looks like you're trying to invoke a `public/subscribe` method, but your 'handler' is called `subscription`. How do you expect these two to match? 
Are you aware of the interface approach where all the event machinery is done for you? All you have to do is declare an interface with an `event` member and JsonRpc will create a proxy for you that implements it so you can 'subscribe' by just adding a handler on the event. And the server just has to include a public event by the same name (usually by implementing the same interface, but that's not required).

If you want to do it manually, you do need your method names and signatures to align. Besides your mismatched method names, I see that your `subscription` method takes no parameters. Yet the code that appears to invoke passes in a `string[] channels` parameter. That would explain the exception you're seeing.
